sklearn/metrics/scorer.py

```diff
@@ -66,6 +67,34 @@ def __repr__(self):
                    "" if self._sign > 0 else ", greater_is_better=False",
                    self._factory_args(), kwargs_string))
 
+    def _get_labels_from_estimator(self, estimator, target_type):
+        """Check if estimator has a classes argument and if scorer has a
+        labels argument and return new kwargs for scorer if label not already
+        set in make score initialization"""
+        est_has_classes = hasattr(estimator, "classes_")
+        scorer_has_labels = "labels" in signature(
```

Comment:
This may all use a bit too much magic. We should really be ensuring that `self._score_func` lives inside `sklearn.metrics` before we do this, since someone else's metric may have different semantics for `labels`... but then the magic is discriminatory. Not great.

I think I have a solution too. Add a parameter `copy_classes` to `make_scorer`. Then we can define relevant scorers with `copy_classes='labels'` (or perhaps just `copy_classes=True`). As a bonus, these cases will necessarily not have `labels` already given.